src/applyContainerQuery.ts

```diff
@@ -17,9 +17,11 @@ export default function<P> (
     },
 
     render() {
-      const props: {[key: string]: any} = assign({ref: this.defineContainerComponent}, this.props);
+      const props: {[key: string]: any} = assign({}, this.props);
       props[propName] = this.state ? this.state.containerQuery : {};
-      return React.createElement(WrappedComponent, props as P);
+      return React.createElement('div', {
```

Comment:
I think we are making a tradeoff here, which I have thought about and I am willing to make. But First I want to make sure there isn't anything I might overlook.

Let's say we have a `WrappedComponent` like this:

```html
<ol>
  <li></li>
</ol>
```

Before this PR, the generated HTML after applying the HOC will be like:

```html
<ol>
  <li></li>
  <div class="erd_scroll_detection_container">...</div>
</ol>
```

(the `div.erd_scroll_detection_container` is added by [element-resize-detector](https://github.com/d6u/react-container-query/blob/c6c8883cbbba95bbd08549180f7d6b2cc0108a4a/package.json#L34) to detect resizes.)

After this PR, the generated HTML after applying the HOC will be like:

```html
<div>
  <ol>
    <li></li>
  </ol>
  <div class="erd_scroll_detection_container">...</div>
</div>
```

I think this approach has pros and cons.

**Pros**

1. The structure of `<ol>`, i.e. the `WrappedComponent` won't be touched. There will be less unexpected behaviors within `WrappedComponent`.
2. Support pure function component (obviously the purpose of this PR)

**Cons**

1. Additional layer of `<div>`. More style might have to be applied, e.g. to make sure `<ol>` take the full height or width of `<div>`. This might confuse people.
2. What if people what HTML element other than DIV? We have to provide a way to custom it.

I love the "structure of `<ol>`, i.e. the `WrappedComponent` won't be touched.", but want a way to address the cons. 

For con 1, I think it is at least as confusing as "touching the internal structure of `WrappedComponent`". We are really not gaining or losing anything, so we should be fine.

For con 2, I think an option to customize the HTML element should be provided (not saying it should be included in this PR, but want to lay down a plan) and default to DIV.

Do you have any suggestion?